tests/test_sanity.py

```diff
@@ -286,6 +286,29 @@ def test_llvm_fastcomp(self):
 
     self.check_working([EMCC, 'tests/hello_world.cpp', '-s', 'ASM_JS=0'], '''Compiler settings are incompatible with fastcomp. You can fall back to the older compiler core, although that is not recommended''')
 
+  def test_check_engine(self):
+    restore()
+    compiler_engine = COMPILER_ENGINE
```

Comment:
please also test both a list and a string - I believe one of my comments shows one of those wasn't handled properly in the `check_engine` code, unless I am confused.